src/opm/output/eclipse/WriteInit.cpp

```diff
@@ -316,19 +317,31 @@ namespace {
                          const ::Opm::UnitSystem&          units,
                          ::Opm::EclIO::OutputStream::Init& initFile)
     {
+#ifdef ENABLE_3DPROPS_TESTING
+        auto active_porv = es.fieldProps().porv();
+        std::vector<double> porv(grid.getCartesianSize());
+        std::size_t active_index = 0;
+        for (auto nGlob    = porv.size(),
+                 globCell = 0*nGlob; globCell < nGlob; ++globCell) {
+            if (grid.cellActive(globCell)) {
+                porv[globCell] = active_porv[active_index];
+                active_index++;
+            } else {
+                porv[globCell] = 0.0;
+            }
```

Comment:
You don't actually need this `else` branch.  Constructing the `porv` vector above will implicitly assign `0.0` to all elements.  In short, this loop is equivalent to
```C++
for (...) {
    if (grid.cellActive(globCell)) {
        // Assign porv
    }
}
```
On the other hand, I don't think you need to loop the global cells at all.  You should be able to write this along the lines of
```C++
std::vector<double> porv(grid.getCartesianSize());  // or porv(grid.getCartesianSize(), 0.0)
std::size_t active_index = 0;
for (const auto& pv : es.fieldProps().porv()) {
    porv[grid.getGlobalIndex(active_index++)] = pv;
}
```

Reply:
Thank you for the suggestions - I will probably implement them.